R-package/R/xgb.DMatrix.R

```diff
@@ -418,26 +450,32 @@ setinfo <- function(object, ...) UseMethod("setinfo")
 #' @rdname setinfo
 #' @export
 setinfo.xgb.DMatrix <- function(object, name, info, ...) {
+  .internal.setinfo.xgb.DMatrix(object, name, info, ...)
+  attr(object, "fields")[[name]] <- TRUE
```

Comment:
Niche case, what happens if I set a Null value?

Reply:
Then it would error out from the C++ side:
```
Error in .internal.setinfo.xgb.DMatrix(object, name, info, ...) :
The length of labels must equal to the number of rows in the input data
```